src/rdo.rs

```diff
@@ -1130,28 +1133,76 @@ fn inter_frame_rdo_mode_decision<T: Pixel>(
         unimplemented!();
       }
     };
-    let mode_set_chroma = ArrayVec::from([luma_mode]);
+    mvs_set.push(mvs);
 
-    luma_chroma_mode_rdo(
-      luma_mode,
+    let tile_rect = ts.tile_rect();
+    let rec = &mut ts.rec.planes[0];
+    let po = tile_bo.plane_offset(rec.plane_cfg);
+    let mut rec_region =
+      rec.subregion_mut(Area::BlockStartingAt { bo: tile_bo.0 });
+
+    luma_mode.predict_inter(
       fi,
-      bsize,
-      tile_bo,
-      ts,
-      cw,
-      rdo_type,
-      cw_checkpoint,
-      &mut best,
-      mvs,
+      tile_rect,
+      0,
+      po,
+      &mut rec_region,
+      bsize.width(),
+      bsize.height(),
       ref_frames_set[i],
-      &mode_set_chroma,
-      false,
-      mode_contexts[i],
-      &mv_stacks[i],
-      AngleDelta::default(),
+      mvs,
     );
+
+    let plane_org = ts.input_tile.planes[0]
+      .subregion(Area::BlockStartingAt { bo: tile_bo.0 });
+    let plane_ref = rec_region.as_const();
+
+    let satd = get_satd(
+      &plane_org,
+      &plane_ref,
+      bsize,
+      fi.sequence.bit_depth,
+      fi.cpu_feature_level,
+    );
+    satds.push(satd);
   });
 
+  let num_modes_rdo = if fi.config.speed_settings.prediction_modes
+    >= PredictionModesSetting::ComplexAll
+  {
+    inter_mode_set.len()
```

Comment:
We probably don't need to perform the predict and satd computations for the complex case, correct?

Reply:
Thanks. It's correct.
I will update to skip these for the complex case.
I will also update to skip the sort by satds for the complex case.